docs/source/testing.rst

```diff
@@ -0,0 +1,316 @@
+want to test whether a single test, which had previously been failing, succeeds
+at last.
+
+Another example for when one would like to test a subset of tests is if the data at
+hand is not available in its entirety. Rather, it could be that one would like
+to run a subset of the test suite against a subsample of the typical dataset.
+
+In this section we present to approaches to do subselection of tests.
+
+Ex-post: subselecting generated tests
+*************************************
+
+Instead of merely running  ``$ pytest specification.py`` one may add pytests's
+``-k`` flag and specify the ``Constraint`` (s) one cares about.
+
+Importantly, every ``Constraint`` object can be identified via a name. If one wants
+to figure out how this string is built, please refer to the implementation of
+:meth:`~datajudge.constraints.base.Constraint.get_description`.
+Otherwise, one could also just run all of the tests once and investigate
+the resulting test report to find the relevant names.
+
+When only caring about the ``UniquesEquality`` constraint in our
+:doc:`twitch example <examples/example_twitch>`.
+one might for instance use the following prefix the filter for it:
+
+.. code-block:: console
+
+  $ pytest twitch_specification.py -k "UniquesEquality::public.twitch_v1"
+
+Ex-ante: Defining categories of tests
+*************************************
+
+Another option to subselect a certain set of tests is by use of
+`pytest markers <https://docs.pytest.org/en/7.1.x/example/markers.html>`_.
+The following is one way of using markers in conjunction with ``datajudge``.
+
+In this particular illustration we'll allow for two markers:
+
+* ``basic``: indicating that only truly fundamental tests should be run
+* ``all``: indicating that any available test should be run
+
+For that matter we'll add a bit of pytest magic to the respective ``conftest.py``.
+
+.. code-block:: python
+  :caption: ``conftest.py``
+
+  def pytest_generate_tests(metafunc):
+      if "basic_constraint" in metafunc.fixturenames:
+           metafunc.parametrize(
+	       "basic_constraint",
+               metafunc.module.get_basic_constraints(),
+               ids=metafunc.module.idfn,
+	   )
+      if "constraint" in metafunc.fixturenames:
+          metafunc.parametrize(
+	      "constraint",
+	      metafunc.module.get_all_constraints(),
+	      ids=metafunc.module.idfn,
+          )
+
+
+Moreover, we'll have to register these markers in pytest's ``pytest.ini`` file.
+You can read more about these files
+`here <https://docs.pytest.org/en/6.2.x/customize.html>`_.
+
+
+.. code-block::
+   :caption: ``pytest.ini``
+
+   [pytest]
+   addopts = --strict-markers
+   markers = basic: basic specification
+	     all: entire specification
+
+Once that is taken care of, one can adapt one's specification as follows:
+
+.. code-block:: python
+  :caption: ``specification.py``
+
+  def get_basic_requirements() -> List[Requirement]:
+      ...
```

Comment:
Is this `...` standing for "add your requirement and constraints"? 

If so, I'd be more explicit since `...` is actually a valid Python syntax.

Reply:
Great point!